framework/include/bcs/GenericIntegratedBC.h

```diff
@@ -26,6 +26,9 @@ class GenericIntegratedBC<true> : public ADIntegratedBC
 public:
   static InputParameters validParams() { return ADIntegratedBC::validParams(); };
   GenericIntegratedBC(const InputParameters & parameters) : ADIntegratedBC(parameters) {}
+
+protected:
+  virtual Real computeQpOffDiagJacobian(unsigned int /*jvar*/) { return 0; }
```

Comment:
make pure virtual?

Reply:
I didn't because not all classes implement it.

Reply:
seems like they should though. How many classes are we talking about?

Reply:
I haven't looked but I don't think computeQpDiagJacobian is ever pure virtual because a lot of classes don't implement it.  I wasn't sure on this, I've only templated materials on is_ad and its been awhile.  I'm following a similar pattern to MatchedValueBC

Reply:
This is also resolved with the up to date pattern. With the new way you can also use `override`. The pattern has already been used in several objects by several other contributors.